app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.html

```diff
@@ -18,16 +18,50 @@
               ng-model="$ctrl.command.payloadJSON"
               ng-change="$ctrl.updatePayload()"/>
 
-    <div class="form-group row slide-in" ng-if="$ctrl.command.invalid">
-      <div class="col-sm-9 col-sm-offset-3 error-message">
-        Error: {{$ctrl.command.errorMessage}}
+      <div class="form-group row slide-in" ng-if="$ctrl.command.invalid">
+        <div class="col-sm-9 col-sm-offset-3 error-message">
+          Error: {{$ctrl.command.errorMessage}}
+        </div>
       </div>
-    </div>
+  </stage-config-field>
+  <stage-config-field label="Custom Headers" help-key="pipeline.config.webhook.customHeaders">
+    <table class="table table-condensed packed">
+      <thead>
+        <tr>
+          <th style="width:40%">Key</th>
+          <th style="width:60%">Value</th>
+          <th class="text-right">Actions</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr ng-repeat="(key,value) in $ctrl.stage.customHeaders">
+          <td>
+            <strong class="small">{{key}}</strong>
+          </td>
+          <td>
+            <input type="text" required ng-model="$ctrl.stage.customHeaders[key]" value="{{value}}"
+                                                                                  class="form-control input-sm"/>
+          </td>
+          <td class="text-right">
+            <a class="small" href ng-click="$ctrl.removeCustomHeader(key)">Remove</a>
+          </td>
+        </tr>
+      </tbody>
+      <tfoot>
+        <tr>
+          <td colspan="7">
+            <button class="btn btn-block btn-sm add-new" ng-click="$ctrl.addCustomHeader()">
+              <span class="glyphicon glyphicon-plus-sign"></span> Add Custom Header
+            </button>
+          </td>
+        </tr>
+      </tfoot>
+    </table>
   </stage-config-field>
   <stage-config-field label="Wait for completion" help-key="pipeline.config.webhook.waitForCompletion">
     <input type="checkbox" class="input-sm" name="waitForCompletion"
-           ng-model="$ctrl.viewState.waitForCompletion"
-           ng-change="$ctrl.waitForCompletionChanged()"/>
+                                            ng-model="$ctrl.viewState.waitForCompletion"
+                                            ng-change="$ctrl.waitForCompletionChanged()"/>
```

Comment:
these whitespace changes probably shouldn't be in here - generally, if attributes for a tag are going to be on a new line, they should be indented to align with the first attribute, i.e. it was good before.